python/ray/experimental/channel/torch_tensor_nccl_channel.py

```diff
@@ -146,58 +135,149 @@ def write(self, value: Any, timeout: Optional[float] = None) -> None:
             raise TypeError(msg) from e
         finally:
             # Pop the tensors that were found during serialization of `value`.
-            tensors_to_send = self.serialization_ctx.reset_tensors([])
+            tensors_to_send, _ = self.serialization_ctx.reset_out_of_band_tensors([])
             # Reset the serialization method to now serialize torch.Tensors
             # normally.
             self.serialization_ctx.set_use_external_transport(False)
 
-        # Send the extracted tensors through a GPU-specific channel.
-        self._gpu_data_channel.write(tensors_to_send)
-        # Send the rest of the data, with placeholders for the extracted
-        # tensors, through a CPU-specific channel.
-        self._cpu_data_channel.write(serialized_cpu_data)
+        # First send the extracted tensors through a GPU-specific channel.
+        self._tensor_data_channel.write(tensors_to_send)
+        # Next send the non-tensor data through a CPU-specific channel. The
+        # data contains placeholders for the extracted tensors.
+        self._non_tensor_data_channel.write(serialized_non_tensor_data)
+
+
+    def write(self, value: Any, timeout: Optional[float] = None) -> None:
+        """
+        Send a value that may contain torch.Tensors that should be sent via
+        external transport.
+
+        This method:
+        1) Serializes `value`. During serialization, all torch.Tensors that are
+        on the default device are extracted and replaced with a unique
+        placeholder. Thus, the serialized value will contain all non-tensor
```

Comment:
I think this is wrong when the tensor is not on the default device (e.g., cpu tensor on gpu actor)